tests/providers/google/cloud/hooks/test_dataproc.py

```diff
@@ -474,7 +474,7 @@ def test_wait_for_job(self, mock_get_job):
             mock.call(region=GCP_LOCATION, job_id=JOB_ID, project_id=GCP_PROJECT),
             mock.call(region=GCP_LOCATION, job_id=JOB_ID, project_id=GCP_PROJECT),
         ]
-        mock_get_job.has_calls(calls)
+        mock_get_job.assert_has_calls(calls, any_order=True)
```

Comment:
Why `any_order` here? Can we order the mock calls correctly so this is not needed?

(Same applies for other usages)

Reply:
@uranusjr @potiuk Agreed, removed it and re-ran the tests, they succeeded. Pushed the changes.

Reply:
What about other usages?

Reply:
@uranusjr Removed them from the remaining tests in the PR. Please let me know if they look good.